### DISCUSSION TITLE
Issue with expo auth implementation

### DISCUSSION THREAD

**Discussion by RajanRV** [ory/hydra#2879] [Q&A] [answered]
We're trying to reconfigure the ory hydra  with azure adb2c sso, but we're not able to configure it as we are new to this.

sharing the configuration we've done for the hydra + azure b2c

domain information: 
    hydra public endpoint: hydra.fbn3.com
    hydra private (admin) endpoint: hydra-admin.fbn3.com
    consent node endpoint: auth.fbn3.com

We're using this configuration in Expo code (React Native): 
`
import { makeRedirectUri, ResponseType } from "expo-auth-session";
import { exchangeCodeAsync, useAuthRequest } from "expo-auth-session";
import * as WebBrowser from "expo-web-browser";

WebBrowser.maybeCompleteAuthSession();

export const discovery = {
  authorizationEndpoint: "https://hydra.fbn3.com/oauth2/auth",
  tokenEndpoint: "https://hydra.fbn3.com/oauth2/token",
  revocationEndpoint: "https://hydra.fbn3.com/oauth2/revoke",
};

export const config = {
  responseType: ResponseType.Code,
  clientId: clientId,
  scopes: scopes,
  // For usage in managed apps using the proxy
  redirectUri: makeRedirectUri({ useProxy: true }),
  usePKCE: false,
};

let [request, response, promptAsync] = useAuthRequest(
    {
      ...config,
    },
    discovery
  );

function authenticate() {
    console.log("AUTHENTICATED!");
    setAuthenticated(true);
  }

  async function login() {
    setLoginStarted(true);
    const result = await promptAsync({ useProxy: true });
    console.log("result from prompt", result);
  }

useEffect(() => {
    login();
}, [])

useEffect(() => {
if (response?.type === "success") {
        const { code } = response.params;
        const auth = await exchangeCodeAsync(
          {
            code: code,
            redirectUri: redirectUri(),
            clientId: clientId,
            scopes: scopes,
          },
          discovery
        );
        await cacheAuthAsync(auth);
        setAccessToken(auth.accessToken);
      }
}, [request, response])

return (
      <View
        style={{
          height: "100%",
          backgroundColor: "#33CA8F",
        }}
      >
        <LoginScreen login={login} showButton={showButton}/>
      </View>
);

`

These are the commands we've used for the hydra image creation in kubernetes. (note: postgresql is already configured)
`
helm install hydra \
    --set 'hydra.config.secrets.system={'$(LC_ALL=C tr -dc 'A-Za-z0-9' < /dev/urandom | base64 | head -c 32)'}' \
    --set 'hydra.config.dsn=postgres://user:password@postgres-postgresql:5432/hydra?sslmode=disable' \
    --set 'hydra.config.urls.self.issuer=https://hydra.fbn3.com/' \
    --set 'hydra.config.urls.login=https://auth.fbn3.com/login' \
    --set 'hydra.config.urls.consent=https://auth.fbn3.com/consent' \
    --set 'hydra.autoMigrate=true' \
    ory/hydra --version 0.21.1

hydra clients create \
    --endpoint https://hydra-admin.fbn3.com \
    --id fbn3staging \
    --scope openid,offline,email,profile \
    --secret secret \
    --callbacks https://auth.fbn3.com/callback \
    -g client_credentials
`

using this config we're getting invalid configuration error while trying to logging in. 
**The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed. The 'redirect_uri' parameter does not match any of the OAuth 2.0 Client's pre-registered redirect urls.**

**Comment by vinckr** [comment, marked as answer]
Hello @RajanRV 

Did you allowlist your redirect URLS? 
https://www.ory.sh/hydra/docs/debug/#oauth-20-authorize-code-flow-fails
